gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2508,51 +2813,87 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
     m_aoFieldDesc.push_back(fieldDesc);
 
     const char *pszLongName = CPLSPrintf("Field %s", poFieldDefn->GetNameRef());
-    status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_LNG_NAME,
-                             strlen(pszLongName), pszLongName);
-    NCDF_ERR(status);
 
     if ( !m_bLegacyCreateMode )
     {
-        char ct_name_carr[NC_MAX_NAME + 1] = {0};
-        status = nc_inq_varname(m_nLayerCDFId, m_writableSGContVarID, ct_name_carr);
-        NCDF_ERR(status);
+        m_poDS->vcdf.nc_put_vatt_text(nVarID, CF_LNG_NAME, pszLongName);
+
+        std::string ct_name(m_layerSGDefn.get_containerName());
+        m_poDS->vcdf.nc_put_vatt_text(nVarID, CF_SG_GEOMETRY, ct_name.c_str());
+    }
 
-        status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_SG_GEOMETRY, strlen(ct_name_carr), ct_name_carr);
+    else
+    {
+        status = nc_put_att_text(m_nLayerCDFId, nVarID, CF_LNG_NAME,
+                                 strlen(pszLongName), pszLongName);
         NCDF_ERR(status);
     }
 
     if( m_bWriteGDALTags )
     {
-        status = nc_put_att_text(m_nLayerCDFId, nVarID, "ogr_field_name",
-                                 strlen(poFieldDefn->GetNameRef()),
-                                 poFieldDefn->GetNameRef());
-        NCDF_ERR(status);
+        if ( !m_bLegacyCreateMode )
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation